tools/tpm2_createprimary.c

```diff
@@ -71,13 +71,17 @@ int setAlg(TPMI_ALG_PUBLIC type,TPMI_ALG_HASH nameAlg,TPM2B_PUBLIC *inPublic)
     // First clear attributes bit field.
     *(UINT32 *)&(inPublic->t.publicArea.objectAttributes) = 0;
     inPublic->t.publicArea.objectAttributes.restricted = 1;
-    inPublic->t.publicArea.objectAttributes.userWithAuth = 1;
+    //check if auth policy needs to be enforced
+    if(E_flag) {
```

Comment:
no need for a branch userWithAuth = flag;

Once flag is a bool type, it should be 0 or 1 by language convention. you could do flag & 0x1 to mask off anything that could accidentally be added.